litestar_ldap/controller.py

```diff
@@ -0,0 +1,98 @@
+"""Routing controller for the LDAP service routes."""
+from typing import Any, List, Optional, Type
+
+import ldap
+from ldap.filter import escape_filter_chars
+from litestar import Controller
+from litestar.status_codes import HTTP_400_BAD_REQUEST, HTTP_404_NOT_FOUND
+from litestar.types.callable_types import Guard
+
+from litestar_ldap.config import LDAPConfig
+
+
+def construct_ldap_controller(
+    ldap_url: str,
+    ldap_tags: Optional[List[str]] = None,
+    ldap_guards: Optional[List[Guard]] = None,
+    ldap_include_in_schema: bool = True,
+) -> Type[Controller]:
+    """Construct the LDAP `controller <litestar.controller.Controller>`.
+
+    Args:
+        ldap_url: The base URL for the LDAP service.
+        ldap_tags: The OpenAPI Schema tags for the LDAP service routes.
+        ldap_guards: The :doc:`guards <litestar:usage/security/guards>` for the LDAP service routes.
+        ldap_include_in_schema: Whether to include the LDAP service routes in the schema.
+
+    Returns:
+        The Litestar controller with the LDAP service routes.
+    """
+    from litestar import Controller, get, post
+    from litestar.datastructures import CacheControlHeader
+
+    class _LDAPController(Controller):
+        """Controller for the LDAP service routes."""
+
+        tags = ldap_tags
+        guards = ldap_guards
+        include_in_schema = ldap_include_in_schema
+
+        @post(
+            operation_id="LoginLDAPUser",
+            name="ldap:login",
+            path=f"{ldap_url}/login",
+            summary="Login via LDAP.",
+            description="Login a user or service account through the LDAP service.",
+        )
+        async def login(self) -> Exception:
+            """Login a user or service account through the LDAP service.
+
+            Returns:
+                The login response.
+            """
+            raise NotImplementedError
+
+        @get(
+            operation_id="GetLDAPUser",
+            name="ldap:getuser",
+            path=f"{ldap_url}/user/{{username:str}}",
+            summary="Get LDAP user.",
+            description="Retrieve a user from the LDAP service.",
+            cache_control=CacheControlHeader(no_store=True),
+        )
+        async def get_ldap_user(self, username: str, ldap_config: LDAPConfig) -> Any:
```

Comment:
you may want to keep this as a sync endpoint until you swap to the asyncio ldap implementation